### PR TITLE
[RESTEASY-2017]:Fix SSE doesn't work with inherited annotations

### PR DESCRIPTION
**Pull request by jimma**

### CODE REVIEW

**Comment by jimma** [comment]
@rsearls  Do you think  https://github.com/resteasy/Resteasy/pull/2348 will fix this intermittent failure too?

**Comment by rsearls** [comment]
I don't know.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/SseAPITest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            eventSource.open();
+            WebTarget sendTarget = client.target(generateURL("/apitest/send"));
+            sendTarget.request().get();
+            boolean result = countDownLatch.await(30, TimeUnit.SECONDS);
+            Assert.assertTrue("Waiting for event to be delivered has timed out.", result);
+        }
+        Assert.assertEquals("One event message was expected.", 1, results.size());
+        Assert.assertTrue("Expected event contains:AnnotationInheritedEvent, but is:" + results.get(0),
+                results.get(0).contains("AnnotationInheritedEvent"));
+    }
```
Comment (asoldano) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/SseAPITest.java:67]: Shouldn't the client (and the event source) be closed before leaving the test?

Comment (jimma) [reply testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/SseAPITest.java:68]: Fixed

**Comment by asoldano** [comment]
@jimma , given we've had 2 more failures on travis, and I've merged #2348, can you please rebase this PR against master before we merge it? That way we can try seeing if the problem goes away or not...

**Comment by asoldano** [comment]
@jimma , please note the SseAPITest being added here has failed twice in Travis in the latest run, we need to ensure it's stable enough before merging the PR. So please try investigating/fixing it.

**Pull request merged**